packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -150,11 +153,29 @@ contract PluginRepo is
     //         contentURI: cid12
     // //  },
 
+    /// @inheritdoc IPluginRepo
+    function updateReleaseMetadata(
```

Comment:
The reason why I asked you to change the order is below function has to be changed and when it will, it's better to have `createVersion, updateMetadata` sequence like this.

If `release == 0` is good check, but not enough.  If I pass `release` as number 55, it will succeed which is not correct. You should be checking if that release was even created.

So, there's a high chance to screw something up big time. After the release == 0 check, we should have:

```
if(release > latestRelease) {
        // revert.
}
```

This avoids to have extra checks here otherwise you would need to do the same checks as in createVersion.

Now, that I think about it, the whole reason why I implemented like that was to always make sure you can't add release 10 after release 8 to be extra safe.  Not sure if this is worth it, but have to think a little bit more once again. for now, let's add the very important revert. 

Though, my comment means dev first has to call createVersion(for the new release) and after that, call `updateMetadata`. Does this sound reasonable to you(the last setence) ? @brickpop

Reply:
NOTE that I know you're doing it vice versa. you got the check in `createVersion`, but problem is, even if the release is not created with `createVersion`, you allow calling update metadata for that release which is not yet created at all with `createVersion` . This will cause the `ReleaseUpdated` event to be thrown and then UI/SDK would be tricked unless they check as well whether or not at least one version exists in order to show on the marketplace the possiblity of installing.

What's your thoughts ? Do you find it logically wrong that you can't call updateMetadata until you call `createVersion` ? it might be little bit logical, but why would you need to update metadata for the release that has 0 builds ?